### DISCUSSION TITLE
Memory increasing during multi server connections

### DISCUSSION THREAD

**Discussion by yUergun** [jchristn/SuperSimpleTcp#103] [Q&A]
Hi,

In my application i am connecting 20 card readers. Readers are working as servers. 
I need to get card readings from all the devices, insert to a database.
I declare a SimpleTcpClient array and connected all of the devices. Everything works fine at the beginning.

My problems:
1- If a device disconnected, my app tries to reconnect every 10 seconds. But in a while (like an hour) memory increases too much and app crashes.

2- If there is no disconnection, memory increases again in 4 hours and app responses so slowly. In this case i can get the read data at the first reading. I have to hold the card the reader few times.

I am using always the same SimpleTcpClient array and releasing objects i have complete using.

Is it possible to flush buffers of SimpleTcpClient objects or am i configuring library wrong?

Connections
for(i=0 ; i<arrDevices.Length ; i++){
	objClient[i] = new SimpleTcpClient(ipPort);
	objClient[i].Keepalive.EnableTcpKeepAlives = true;
	objClient[i].Events.Connected += Events_Connected;
	objClient[i].Events.Disconnected += Events_Disconnected;
	objClient[i].Events.DataReceived += Events_DataReceived;
	objClient[i].Settings.ConnectTimeoutMs = 1000;
	objClient[i].Settings.MutuallyAuthenticate = false;
	objClient[i].Settings.AcceptInvalidCertificates = true;

	objClient[i].ConnectWithRetries(1000);
}

Sending Data 
objClient[i].SendAsync(strRequest);

**Comment by jchristn** [comment]
Are you disposing ```SimpleTcpClient``` before reusing?

**Reply by yUergun** [reply]
Thanks for your reply and support.

Yes. I am disposing in the disconnected event.

    private void Events_Disconnected(object sender, SimpleTcp.ClientDisconnectedEventArgs e)
    {
        this.Invoke((MethodInvoker)delegate {
            objClient[dictReader[e.IpPort].ReaderIndex].Dispose();

            strLogMessage = $"Disconnected Line:{e.IpPort} Reason:{e.Reason}";
            logConsole.Text += strLogMessage + Environment.NewLine;
        });
    }
Even if i use Dispose() wrong, disposing only solves my first problem. App is still slowing down in 3-4 hours.

**Reply by jchristn** [reply]
Ok.  Is there any way to produce a sample solution that reproduces this problem?
